tensorboard/webapp/widgets/data_table/data_table_test.ts

```diff
@@ -789,4 +798,125 @@ describe('data table', () => {
     expect(dataElements[3].nativeElement.innerText).toBe('1');
     expect(dataElements.length).toBe(4);
   });
+
+  describe('delete column button', () => {
+    it('emits removeColumn event when delete button clicked', () => {
+      const fixture = createComponent({
+        headers: [
+          {
+            type: ColumnHeaderType.VALUE,
+            name: 'value',
+            displayName: 'Value',
+            enabled: true,
+          },
+          {
+            type: ColumnHeaderType.RUN,
+            name: 'run',
+            displayName: 'Run',
+            enabled: true,
+          },
+          {
+            type: ColumnHeaderType.STEP,
+            name: 'step',
+            displayName: 'Step',
+            enabled: true,
+          },
+          {
+            type: ColumnHeaderType.RELATIVE_TIME,
+            name: 'relativeTime',
+            displayName: 'Relative',
+            enabled: true,
+          },
+        ],
+      });
+      fixture.componentInstance.hparamsEnabled = true;
+      fixture.detectChanges();
+      const headerElements = fixture.debugElement.queryAll(
+        By.css('.header > .col')
+      );
+
+      headerElements[3]
+        .queryAll(By.css('mat-icon'))[1]
+        .triggerEventHandler('click', {});
```

Comment:
nit: for the "act" part, it's line 834 and line 838. Let's group them together. (i.e. a newline between line 833-834
```
      fixture.detectChanges();

      const headerElements = fixture.debugElement.queryAll(
        By.css('.header > .col')
      );
      headerElements[3]
        .queryAll(By.css('mat-icon'))[1]
        .triggerEventHandler('click', {});
```